idl/internal/docstring.go

```diff
@@ -0,0 +1,138 @@
+package internal
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	_docstringOpen  = "/**"
+	_docstringClose = "*/"
+)
+
+// CleanupDocstring takes a docstring in the form,
+//
+//  /**
+//   * foo bar
+//   */
+//
+// And returns,
+//
+//  foo bar
+func CleanupDocstring(s string) string {
+	lines := strings.Split(s, "\n")
+	dedent(lines, true /* skipFirstIfUnindented */)
+
+	switch len(lines) {
+	case 0:
+		return s
+	case 1:
+		// /** foo */
+		s := lines[0]
+		s = strings.TrimPrefix(s, _docstringOpen)
+		s = strings.TrimSuffix(s, _docstringClose)
+		return strings.TrimSpace(s)
+	}
+
+	if len(lines) > 0 {
```

Comment:
We must have `len(lines) > 0` if we got to this point

Reply:
Good catch with this and the other comment. A bunch of code got moved around but I forgot to remove the checks that were now redundant.